include/dxc/Support/Path.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "llvm/ADT/StringRef.h"
+#include <string>
+
+namespace hlsl {
+
+template <typename CharTy>
+bool IsAbsoluteOrCurDirRelativeShared(const CharTy *Path) {
```

Comment:
I spent a while trying to figure out what this is meant to do.  The "Shared" threw me quite a bit because I thought it was something to do with "CurDirRelativeShared", but I think the "Shared" just means that this same function is called by both the W and A versions?

In the absence of any unit tests, a comment giving some example inputs and the expected results would be helpful to understand and validate this.

Also, something somewhere explaining the overall strategy of path normalization would be helpful.  From the code we can see the somewhat arbitrary set of things that happen without really understanding why any of it is done.

Reply:
I changed the "Shared" to "Impl", just like the Normalize function below.

Also added explanation and examples in the Normalize function below.